src/builtins/test.rs

```diff
@@ -1028,15 +1030,32 @@ pub fn test(parser: &Parser, streams: &mut IoStreams, argv: &mut [&wstr]) -> Opt
         .collect();
     let args: &[WString] = &args;
 
-    if argc == 0 {
-        return STATUS_INVALID_ARGS; // Per 1003.1, exit false.
-    } else if argc == 1 {
-        // Per 1003.1, exit true if the arg is non-empty.
-        return if args[0].is_empty() {
-            STATUS_CMD_ERROR
-        } else {
-            STATUS_CMD_OK
-        };
+    if feature_test(FeatureFlag::test_require_arg) {
+        if argc == 0 {
+            streams.err.appendln(wgettext_fmt!(
+                "%ls: Expected at least one argument",
+                program_name
+            ));
+            builtin_print_error_trailer(parser, streams.err, program_name);
+            return STATUS_INVALID_ARGS;
```

Comment:
Good change.

One alternative is to keep returning the same exit code as today, but only print the error. This way we wouldn't really need a feature flag.  I'm fine with either.

Reply:
That's not a bad idea. That could be our "depreciation phase".

Reply:
correction, we'd still want a feature flag